orangecontrib/argument/miner/chunker.py

```diff
@@ -23,170 +19,273 @@
-        )
-        df_temp = df_temp.groupby(by="argument_id", as_index=False).agg(
-            {"embed": rank_in_argument}
-        )
-        ranks = df_temp["embed"].tolist()
-        ranks = list(itertools.chain(*ranks))
-        self.df_chunks["rank"] = ranks
 
-    def get_chunk_table(self) -> pd.DataFrame:
-        """Get full info table of chunks.
+def get_chunk_polarity_score(chunks: List[str]):
+    """Compute polarity score of each chunk in the given list.
 
-        This table will include the following columns: chunk, argument_id,
-        topic, topic_prob, rank, polarity_score.
-        """
-        self.chunk()
-        self.chunk_topic()
-        self.chunk_rank()
-        self.chunk_polarity_score()
-        return copy.deepcopy(self.df_chunks)
+    The polarity score is a float within the range [-1.0, 1.0], where 0 means neutral, + means positive, and - means negative.
 
-    def get_topic_table(self) -> pd.DataFrame:
-        """Get topic info table.
+    Args:
+        chunks (List[str]): chunk list
 
-        This table will include the following columns: topic, name, count,
-        keywords, keyword_scores.
-        """
-        assert (
-            "topic" in self.df_chunks.columns
-        ), "Should do topic modeling before getting topic info table!"
-        return self.topic_model.get_topic_table()
+    Returns:
+        List[float]: polarity scores of the given chunks
+    """
+    p_scores = []
+    for chunk in chunks:
+        p_scores.append(TextBlob(chunk).sentiment.polarity)
+    return p_scores
+
+
+def get_chunk_topic(chunks: List[str]):
+    """Get topic information and embedding vectors of chunks via topic modeling.
+
+    Args:
+        chunks (List[str]): chunk list.
+
+    Returns:
+        List[int]: topic ids of chunks.
+        np.ndarray: embedding vectors of chunks.
+        pd.DataFrame: Table of topic information.
+    """
+    topic_model = TopicModel()
+    topic_model.init_models()
+    topics = topic_model.fit_transform_reduced(chunks)
+    embeds = topic_model.get_doc_embed()
+    df_topics = topic_model.get_topic_table()
+    return topics, embeds, df_topics
+
+
+def get_chunk_rank(arg_ids: List[int], embeds: np.ndarray):
+    """In each argument, comput rank of chunks within.
+
+    Rank can be understand as importance of chunks. This way, this function compute the relative importance of chunks within arguments they belongs to. This is done by applying Pagerank algorithm, where similarity is computed as the cosine similarity of chunk embedding vectors.
+
+    Args:
+        arg_ids (List[int]): ids of arguments that chunks belongs to.
+        embeds (np.ndarray): embedding vectors of chunks.
 
+    Returns:
+        List[float]: rank of chunks
+    """
 
-class ArgumentTopic(BERTopic):
+    def rank_in_argument(embeds):
+        """Compute pagerank of embedding vectors."""
+        embeds = list(embeds)
+        embeds = torch.tensor(embeds)
```

Comment:
`embeds` as input is not an numpy array, but a pandas series, as function `rank_in_argument` is called within `agg`, so it anyway has to be transferred to some other formats, either tensor or numpy array. Also, I can't remind any specific reason why I use torch here (maybe because I had read something related to torch at that time point). But indeed, I tried to implement the same logic with numpy and it's faster. Eventually, I switch to the equalvalent numpy implementation.